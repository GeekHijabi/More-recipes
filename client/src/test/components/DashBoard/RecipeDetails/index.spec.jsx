import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ConnectedMyRecipe, { MyRecipe }
  from '../../../../js/components/DashBoard/RecipeDetails/index';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const props = {
  onViewRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiUpVoteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiDownVoteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apifavoriteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
  apiRecipeViewCount: jest.fn(() => Promise.resolve({ ok: true })),
  recipe: {},
  match: {
    params: {
      id: 2
    }
  },
  views: 0,
  history: {
    push: () => Promise.resolve({ ok: true }),
  }

};


describe('MyRecipe snapshot', () => {
  it('should render correctly', () => {
    const recipeAdmin = shallow(<MyRecipe {...props} />);
    expect(recipeAdmin).toMatchSnapshot();
  });
});

describe('RecipeAdmin component', () => {
  describe('handleChangeTab', () => {
    it('should change the tab', () => {
      const wrapper = shallow(<MyRecipe {...props} />);
      //   wrapper.setState({ activeTab: 'tab' });
      wrapper.instance().handleChangeTab(3);
      expect(wrapper.state().activeTab).toEqual(3);
    });
  });
  describe('handleUpvotes', () => {
    it('should handle the upvote', () => {
      const wrapper = shallow(<MyRecipe {...props} />);
      wrapper.instance().handleupvote();
      expect(wrapper.instance().props.apiUpVoteRecipe).toBeCalled();
    });
  });

  describe('handleDownvotes', () => {
    it('should handle downvote', () => {
      const wrapper = shallow(<MyRecipe {...props} />);
      wrapper.instance().handledownvote();
      expect(wrapper.instance().props.apiDownVoteRecipe).toBeCalled();
    });
  });

  describe('handleFavorite', () => {
    it('should handle favorite', () => {
      const wrapper = shallow(<MyRecipe {...props} />);
      wrapper.instance().handlefavorite();
      expect(wrapper.instance().props.apifavoriteRecipe).toBeCalled();
    });
  });

  describe('onViewRecipe', () => {
    it('should throw error if wrong route is called', () => {
      const rejectProps = {
        onViewRecipe: jest.fn(() => Promise.reject(new Error('fail'))),
        apiUpVoteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
        apiDownVoteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
        apifavoriteRecipe: jest.fn(() => Promise.resolve({ ok: true })),
        apiRecipeViewCount: jest.fn(() => Promise.resolve({ ok: true })),
        recipe: {},
        match: {
          params: {
            id: 2
          }
        },
        views: 0,
        history: {
          push: () => Promise.resolve({ ok: true }),
        }
      };
      const wrapper = shallow(<MyRecipe {...rejectProps} />);
      expect(wrapper.instance().props.onViewRecipe).toBeCalled();
    });
  });

  describe('Connected MyRecipe component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        recipe: {
          recipe: {},
          views: 0
        },
      });
      const wrapper = shallow(<ConnectedMyRecipe store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
