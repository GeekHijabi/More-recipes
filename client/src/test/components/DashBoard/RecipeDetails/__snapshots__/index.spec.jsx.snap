// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipe snapshot should render correctly 1`] = `
<div>
  <withRouter(Connect(RecipeHeader)) />
  <main
    className="row"
    page="details"
  >
    <div
      className="dual col-xs-12 col-sm-6 col-md-6 col-lg-7"
    >
      <img
        className="food"
        src={Object {}}
      />
      <h4>
        <span
          className="food_name"
        />
      </h4>
      <div
        className="food_vote"
      >
        <span
          className="vote_type"
        >
          <i
            className="fa fa-thumbs-o-up fa-2x"
            data-toggle="tool-tip"
            onClick={[Function]}
            role="button"
            tabIndex="-1"
            title="upvote recipe"
          />
          <span>
            upvote
          </span>
          <span
            className="detail-value"
          />
        </span>
        <span
          className="vote_type"
        >
          <i
            className="fa fa-thumbs-o-down fa-2x"
            data-toggle="tool-tip"
            onClick={[Function]}
            role="button"
            tabIndex="-1"
            title="downvote recipe"
          />
          <span>
            downvote
          </span>
          <span
            className="detail-value"
          />
        </span>
        <span
          className="vote_type"
        >
          <i
            className="fa fa-heart fa-2x"
            data-toggle="tool-tip"
            onClick={[Function]}
            role="button"
            tabIndex="-1"
            title="add to favorites"
          />
          <span>
            favorite
          </span>
          <span
            className="detail-value"
          />
        </span>
        <span
          className="vote_type"
        >
          <i
            className="fa fa-eye fa-2x"
            data-toggle="tool-tip"
            title="number of views"
          />
          <span>
            view
          </span>
          <span
            className="detail-value"
          >
            0
          </span>
        </span>
      </div>
    </div>
    <div
      className="dual col-xs-12 col-sm-6 col-md-6 col-lg-5"
    >
      <Tabs
        activeTab={
          Object {
            "downvotes": false,
            "isActive": true,
            "name": "Descriptions",
            "upvotes": false,
          }
        }
        changeTab={[Function]}
        isActive={false}
      />
      <Content
        activeTab={
          Object {
            "downvotes": false,
            "isActive": true,
            "name": "Descriptions",
            "upvotes": false,
          }
        }
        description=""
        ingredients=""
        recipe={Object {}}
        recipeId={0}
        reviewedRecipe={Object {}}
      />
    </div>
  </main>
  <Footer />
</div>
`;
